templates/prometheus-servicemonitor.yaml

```diff
@@ -0,0 +1,37 @@
+{{ template "vault.mode" . }}
+{{ if .Values.prometheus.operator.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "vault.fullname" . }}
+  labels:
+    helm.sh/chart: {{ include "vault.chart" . }}
+    app.kubernetes.io/name: {{ include "vault.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    {{- toYaml .Values.prometheus.operator.serviceMonitor.selector | nindent 4 }}
```

Comment:
Should `selector` here be called `labels` instead?

Reply:
I would prefer to leave it as selector, as it needs to match up with the serviceMonitorSelector in the Prometheus Operator configuration. See here: https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md

Happy to change it if it's still preferred though.